core/src/main/java/hudson/model/Actionable.java

```diff
@@ -165,7 +178,8 @@ public void replaceAction(@Nonnull Action a) {
     }
 
     /**
-     * Remove an action.
+     * Remove an action. Note calls to {@link #getAllActions()} that happen before calls to this method may not see the
+     * update.
```

Comment:
Not just that. `getAllActions()` *after* this call may still see that action, if it was added by a transient factory.